panoply/resources.py

```diff
@@ -12,5 +12,32 @@ class Resource(TypedDict):
     id: str
     title: str
     fields: Optional[List[Field]]
+    available: Optional[bool]
     required: Optional[bool]
     requires: Optional[List[str]]
+
+
+def list_resources(resources: List[Resource]) -> Optional[List[Dict]]:
+    if resources:
+        return [
+            {"name": resource["title"],
+             "value": resource["id"],
+             "disabled": not resource.get("available", True),
+             "required": resource.get("required", False),
+             "requires": resource.get("requires", [])}
+            for resource in resources]
+    else:
+        return
+
+
+def list_fields(fields: List[Field]) -> Optional[List[Dict]]:
+    if fields:
+        return [
+            {"name": f"{field['name']} [{field['type']}]" if field.get("type") else field["name"],
+             "value": field["name"],
+             "type": field.get("type"),
+             "is_mandatory": field.get("is_mandatory", False),
+             "disabled": not field.get("is_available", True)}
```

Comment:
@zoriana2021, @ilia-panoply, same, we can leave only `name` and `type` and render title (`f"{field['name']} [{field['type']}]" if field.get("type") else field["name"]`) on UI so we won't need to send more data here. I hope it's not late to change it.

Reply:
@kirill-panoply 
For UI we need at least `name` and `value`. And also `disable`.

Regarding other fields, I prefer that we have this information in source object so there is no need to get this information again when we need it.

Reply:
Okay, got it, thought we'll use different param for that.